snapfaas/src/sched/rpc.rs

```diff
@@ -0,0 +1,107 @@
+use std::net::{TcpStream, SocketAddr};
+use std::thread;
+use std::collections::HashMap;
+use std::collections::hash_map::DefaultHasher;
+use std::hash::{Hash, Hasher};
+use serde::{Serialize, Deserialize};
+
+use super::Error;
+use super::message;
+use super::message::{Request, Response};
+use super::message::request::Kind as ReqKind;
+
+/// RPC calls
+#[derive(Debug)]
+pub struct Scheduler {
+    _sock_addr: SocketAddr, // reconnect
+    stream: TcpStream,
+}
+
+impl Scheduler {
+    pub fn new(addr: String) -> Self {
+        let stream = TcpStream::connect(&addr).unwrap();
+        let _sock_addr = addr.parse().unwrap();
+        Scheduler { _sock_addr, stream }
+    }
+
+    /// This method is for workers to retrieve a HTTP request, and
+    /// it is supposed to block if there's no further HTTP requests
+    pub fn get(&mut self) -> Result<Response, Error> {
+        // avoid using unstable #![feature(thread_id_value)]
+        let thread_id = {
```

Comment:
task_id?

Reply:
No this is the thread id used to identify worker threads for the scheduler. This is not useful right now but if we want worker threads to share RPC connections, this is useful.

Reply:
I think worker threads on the same worker machine are sharing one RPC connection? For one, we use `Arc<Mutex<Scheduler>>` in vm.rs